code/obj/item.dm

```diff
@@ -875,7 +875,20 @@
 		STOP_TRACKING_CAT(TR_CAT_BURNING_ITEMS)
 	burning_last_process = src.burning
 
+/// Call this proc inplace of attack_self(...)
+/obj/item/proc/AttackSelf(mob/user as mob)
+	SHOULD_NOT_OVERRIDE(1)
+	if(SEND_SIGNAL(src, COMSIG_ITEM_ATTACK_SELF, user))
+		return
+	src.attack_self(user)
```

Comment:
I'd personally make this be `. = src.attack_self(user)` because I feel like a proper wrapper should be transparent w.r.t. return values even if they are currently unused but it likely doesn't matter at all.

Reply:
Agreed